src/acvm_interop/proof_system.rs

```diff
@@ -21,58 +19,77 @@ impl ProofSystemCompiler for Barretenberg {
         )?)
     }
 
-    fn black_box_function_supported(&self, opcode: &BlackBoxFunc) -> bool {
+    fn supports_opcode(&self, opcode: &Opcode) -> bool {
         match opcode {
-            BlackBoxFunc::AND
-            | BlackBoxFunc::XOR
-            | BlackBoxFunc::RANGE
-            | BlackBoxFunc::SHA256
-            | BlackBoxFunc::Blake2s
-            | BlackBoxFunc::Keccak256
-            | BlackBoxFunc::ComputeMerkleRoot
-            | BlackBoxFunc::SchnorrVerify
-            | BlackBoxFunc::Pedersen
-            | BlackBoxFunc::HashToField128Security
-            | BlackBoxFunc::EcdsaSecp256k1
-            | BlackBoxFunc::FixedBaseScalarMul => true,
-
-            BlackBoxFunc::AES => false,
+            Opcode::Arithmetic(_) => true,
+            Opcode::Directive(_) => true,
+            Opcode::Block(_) => false,
```

Comment:
We don't support `Block`? cc'ing @guipublic

Reply:
It doesn't matter that much right now since the current Noir code will just use the block opcode anyways.

It will matter more depending on how 1282 is implemented. Block is a bit weird because its sort of like the fallback implementation for ROM/RAM

Reply:
Sounds like this PR is good to go then?

Reply:
Yeah if that was the only blocker, then we can tenatively merge